### ISSUE TITLE
Reproducing results for "A benchmark for surface reconstruction"

### ISSUE THREAD

**Issue by raphaelsulzer** [amosgropp/IGR#2]
Hi,

thank you for making this great work available here!

I'm wondering how I could reproduce your results for the five shapes of Berger et al.'s "A benchmark for surface reconstruction" (section 7.1 in your paper).
Is there a pretrained model available for this task?

Kind regards

**Comment by amosgropp** [comment]
Hi Raphael!
Unfortunately, we didn't log a pre-trained model for this task.
If you want you can reproduce these results by training the network yourself as described in the README in the surface reconstruction section.
The hyper-parameters for this task are specified in the appendix.
We used the git of DGP for comparisons and downloading the benchmark:
https://github.com/fwilliams/deep-geometric-prior
Good luck!
Amos

**Comment by raphaelsulzer** [comment]
Hi Amos,
thank you for your help!
In fact I would like to train the network on 4 different scanner settings for each of the 5 shapes of the benchmark.

Any idea how I could speed up this task of training the network 20 times?

I managed to start a training process with the provided config file. If I am not mistaking, it should be the same settings which you used in the paper for this benchmark. However, training only on one single shape will already take several hours.

**Comment by amosgropp** [comment]
You can try to speed it up with a smaller network or different learning rate.

I would try to schedule a higher learning rate in the begging and slow it down - currently, it is a constant learning rate.

Make sure you use the normals of the inputs - it would lead to better results and also speed convergence.

Good luck

**Comment by raphaelsulzer** [comment]
Hi again,

thanks for the advice!

> Make sure you use the normals of the inputs - it would lead to better results and also speed convergence.

How exactly do I do that?

I saved my data as a .npz file which includes an array like [point_x, point_y, point_z, normal_x, normal_y, normal_z].

Then I replaced line 58 in utils/general.py with,
`point_set = np.load(file_name)`
`point_set = torch.tensor(point_set['arr_0']).float()`.

Thus, my point set should now have normals. However, I am still getting the following message during the beginning of training:

> face_normals incorrect shape, ignoring!

Obviously my point cloud does not have face_normals, but I am not sure where this error is coming from to investigate further.

Kind regards


**update:**
actually, I just realised that the normal_loss is much smaller than before (when I simply used the .ply loader), so I guess the network is probably using the provided normals now.
